### DISCUSSION TITLE
'Blocked executable in the ESP' no longer an issue?!

### DISCUSSION THREAD

**Discussion by laur89** [fwupd/fwupd#5909] [Q&A] [answered]
The old issue of unused image in /boot partition blocking upgrade (see https://github.com/fwupd/fwupd/issues/5035) suddenly stopped being [a problem for me](https://github.com/fwupd/fwupd/issues/5035#issuecomment-1259445952) - `fwupd` now updates without raising any errors, and my `/boot/efi` looks exactly the same when upgrade command first stopped working.

Just curious as to what has changed? Was some sort of workaround implemented?

Edit: attaching `/boot/efi/EFI/BOOT/BOOTX64.EFI` in question:
[BOOTX64.EFI.zip](https://github.com/fwupd/fwupd/files/11768689/BOOTX64.EFI.zip)

**Comment by hughsie** [comment]
Did the checksum change? Did Windows update that file, or maybe shim was updated in your distribution?

**Reply by laur89** [reply]
Nope - I've ran the same single OS since I got the computer, and last mod time is unchanged:
```
$ sudo ls -lt /boot/efi/EFI/BOOT
-rwx------ 1 root root 1322936 Feb  1  2021 BOOTX64.EFI
-rwx------ 1 root root 1207944 Feb  1  2021 fbx64.efi
-rwx------ 1 root root 1660272 Feb  1  2021 grubx64.efi
```
Which algo is used for checksum? If it's sha256, then it _has_ changed, but I can't see how or why (and note the last mod date is still in 2021!):
```
$ sudo sha256sum /boot/efi/EFI/BOOT/BOOTX64.EFI
599a102b6445fa88392b8c85a31d80ece950624219d846affbfb7131d4bf550b  /boot/efi/EFI/BOOT/BOOTX64.EFI
/--/
af79b14064601bc0987d4747af1e914a228c05d622ceda03b7a4f67014fee767  <- sum in the error message
```

**Reply by hughsie** [reply]
The error message checksum is the Authenticode checksum, not the SHA256 checksum unfortunately.

> sudo sha256sum /boot/efi/EFI/BOOT/BOOTX64.EFI
> 599a102b6445fa88392b8c85a31d80ece950624219d846affbfb7131d4bf550b  /boot/efi/EFI/BOOT/BOOTX64.EFI

That should have been detected! That's this line in the dbx:

    599A102B6445FA88392B8C85A31D80ECE950624219D846AFFBFB7131D4BF550B,AF79B14064601BC0987D4747AF1E914A228C05D622CEDA03B7A4F67014FEE767,Signed_14173467011297444/shim64-bit.efi,64-bit,Debian,CVE-2020-14372; CVE-2020-25632; CVE-2020-25647; CVE-2020-27749; CVE-2020-27779; CVE-2021-3418; CVE-2021-20225; CVE-2021-20233,Apr-21

**Reply by laur89** [reply]
Possible regression?
```
fwupdmgr --version
compile   org.freedesktop.fwupd         1.8.12
compile   com.hughsie.libxmlb           0.3.10
compile   com.hughsie.libjcat           0.1.9
runtime   org.freedesktop.fwupd-efi     1.4
compile   org.freedesktop.gusb          0.3.10
runtime   com.dell.libsmbios            2.4
runtime   org.freedesktop.gusb          0.3.10
runtime   org.kernel                    6.1.0-9-amd64
runtime   org.freedesktop.fwupd         1.8.12
```

**Reply by hughsie** [reply]
can you attach your /boot/efi/EFI/BOOT/BOOTX64.EFI to this report please -- you might need to zip it with a password to allow github to attach it.

**Reply by laur89** [reply]
Attached to OP.

**Reply by hughsie** [reply, marked as answer]
Thanks. Putting that file in my ESP I get:

    Blocked executable in the ESP, ensure grub and shim are up to date: /run/media/root/1E3C-C27B/EFI/Microsoft/Boot/BOOTX64.EFI Authenticode checksum [af79b14064601bc0987d4747af1e914a228c05d622ceda03b7a4f67014fee767] is present in dbx

I tried that on fwupd 1.9.2 and 1.8.16

Did you perhaps install the dbx manually with `fwupdtool --force`?

I also tried with 1.8.12 but was getting `Not compatible with org.freedesktop.fwupd version 1.8.12, requires >= 1.8.14` with `DBXUpdate-20230314-x64.cab` -- I suspect fwupd was installing the 3rd to newest dbx (which is still newer than you had before!) which didn't have the dbx authenticode checksums for your BOOTX64.EFI -- i.e. fwupd did the right thing. If it didn't do the right thing your system wouldn't have booted after applying the dbx update :)

I think if you update to fwupd >= 1.8.14 you'll get one further dbx update (which will fail to install until you update your shim package) and then yet another if you update to fwupd 1.9.1.

**Reply by laur89** [reply]
Thanks for having a look.

> Did you perhaps install the dbx manually with fwupdtool --force

Have been interacting with fwupd only [via own shell abstraction](https://github.com/laur89/dotfiles/blob/9e267e0f7188741d60f3e263f2989608d052099f/.bootstrap/install_system.sh#L1972-L1992) which never uses fwupdtool.

> I suspect fwupd was installing the 3rd to newest dbx

Does this imply my fwupd version was downgraded at one point?

**Reply by hughsie** [reply]
> Does this imply my fwupd version was downgraded at one point?

I don't think so, maybe the opposite. i.e. 1.7.x might have been refusing to install *any dbx* update because it wanted to install the newest one, but 1.8.12 did a better job of getting you closer to where you need to be by looking at the requirements before filtering.
